azure-mgmt-containerinstance/src/main/java/com/microsoft/azure/management/containerinstance/implementation/ContainerGroupImpl.java

```diff
@@ -0,0 +1,314 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.containerinstance.implementation;
+
+import com.microsoft.azure.management.apigeneration.LangDefinition;
+import com.microsoft.azure.management.containerinstance.Container;
+import com.microsoft.azure.management.containerinstance.ContainerGroup;
+import com.microsoft.azure.management.containerinstance.ContainerGroupNetworkProtocol;
+import com.microsoft.azure.management.containerinstance.ImageRegistryCredential;
+import com.microsoft.azure.management.containerinstance.OperatingSystemTypes;
+import com.microsoft.azure.management.containerinstance.Port;
+import com.microsoft.azure.management.containerinstance.Volume;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.GroupableParentResourceImpl;
+import rx.Observable;
+import rx.functions.Func1;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Implementation for ContainerGroup and its create interfaces.
+ */
+@LangDefinition
+public class ContainerGroupImpl
+    extends
+    GroupableParentResourceImpl<
+                ContainerGroup,
+                ContainerGroupInner,
+                ContainerGroupImpl,
+                ContainerInstanceManager>
+    implements ContainerGroup,
+    ContainerGroup.Definition {
+
+    private Map<String, Container> containers;
+    private Map<String, Volume> volumes;
+    private List<String> imageRegistryServers;
+    private int[] externalTcpPorts;
+    private int[] externalUdpPorts;
+
+    protected ContainerGroupImpl(String name, ContainerGroupInner innerObject, ContainerInstanceManager manager) {
+        super(name, innerObject, manager);
+    }
+
+    @Override
+    protected void beforeCreating() {
+    }
+
+    @Override
+    protected Observable<ContainerGroupInner> createInner() {
+        return this.manager().inner().containerGroups().createOrUpdateAsync(this.resourceGroupName(), this.name(), this.inner());
+    }
+
+    @Override
+    protected void afterCreating() {
+        initializeChildrenFromInner();
+    }
+
+    @Override
+    protected void initializeChildrenFromInner() {
+        // Getting the container instances
+        if (this.inner().containers() != null && this.inner().containers().size() > 0) {
+            this.containers = new HashMap<>();
```

Comment:
what if `.refresh()` results in `.containers()` being null or empty --- where is `this.containers` initialized?  Shouldn't `this.containers = new HashMap<>()` be before the if?

Reply:
fixed to return "null" in that case.